### ISSUE TITLE
Code generation fails in JDK 19

### ISSUE THREAD

**Issue by dwragge** [immutables/immutables#1395]
Hey there, running with JDK 19-ea, compilation fails with the below stack trace. Any ideas on this one? It looks like some internal JDK compiler class has changed type.

Cheers,
Dylan

```
openjdk 19-ea 2022-09-20
OpenJDK Runtime Environment (build 19-ea+28-2110)
OpenJDK 64-Bit Server VM (build 19-ea+28-2110, mixed mode, sharing)
```

```
[ERROR] error: org.immutables.value.internal.$processor$.$Processor threw java.lang.ClassCastException: class org.immutables.value.internal.$processor$.meta.$CachingElements$CachingTypeElement cannot be cast to class com.sun.tools.javac.code.Symbol (org.immutables.value.internal.$processor$.meta.$CachingElements$CachingTypeElement is in unnamed module of loader java.net.URLClassLoader @719450d6; com.sun.tools.javac.code.Symbol is in module jdk.compiler of loader 'app')
[ERROR]         at jdk.compiler/com.sun.tools.javac.model.JavacElements.getFileObjectOf(JavacElements.java:732)
[ERROR]         at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
[ERROR]         at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:1006)
[ERROR]         at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
[ERROR]         at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
[ERROR]         at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:575)
[ERROR]         at java.base/java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260)
[ERROR]         at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:616)
[ERROR]         at jdk.compiler/com.sun.tools.javac.processing.JavacFiler.originatingFiles(JavacFiler.java:531)
[ERROR]         at jdk.compiler/com.sun.tools.javac.processing.JavacFiler.createSourceOrClassFile(JavacFiler.java:509)
[ERROR]         at jdk.compiler/com.sun.tools.javac.processing.JavacFiler.createSourceFile(JavacFiler.java:435)
[ERROR]         at org.immutables.value.internal.$generator$.$Output$SourceFile.complete($Output.java:313)
[ERROR]         at org.immutables.value.internal.$generator$.$Output$7.invoke($Output.java:177)
[ERROR]         at org.immutables.value.internal.$generator$.$Intrinsics.$($Intrinsics.java:96)
[ERROR]         at org.immutables.value.internal.$processor$.$Generator_Immutables._t0__generate($Generator_Immutables.java:20)
[ERROR]         at org.immutables.value.internal.$processor$.$Generator_Immutables$FragmentDispatch.run($Generator_Immutables.java:19603)
[ERROR]         at org.immutables.value.internal.$generator$.$Templates$Fragment.invoke($Templates.java:248)
[ERROR]         at org.immutables.value.internal.$generator$.$AbstractGenerator.invoke($AbstractGenerator.java:57)
[ERROR]         at org.immutables.value.internal.$processor$.$Processor.process($Processor.java:77)
[ERROR]         at org.immutables.value.internal.$generator$.$AbstractGenerator.process($AbstractGenerator.java:87)
[ERROR]         at org.immutables.processor.ProxyProcessor.process(ProxyProcessor.java:72)
[ERROR]         at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:1018)
[ERROR]         at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:934)
[ERROR]         at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1262)
[ERROR]         at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1377)
[ERROR]         at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1234)
[ERROR]         at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:916)
[ERROR]         at jdk.compiler/com.sun.tools.javac.main.Main.compile(Main.java:317)
[ERROR]         at jdk.compiler/com.sun.tools.javac.main.Main.compile(Main.java:176)
[ERROR]         at jdk.compiler/com.sun.tools.javac.Main.compile(Main.java:82)
[ERROR]         at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
[ERROR]         at java.base/java.lang.reflect.Method.invoke(Method.java:578)
[ERROR]         at org.codehaus.plexus.compiler.javac.JavacCompiler.compileInProcess0(JavacCompiler.java:626)
[ERROR]         at org.codehaus.plexus.compiler.javac.JavacCompiler.compileInProcessWithProperClassloader(JavacCompiler.java:607)
[ERROR]         at org.codehaus.plexus.compiler.javac.JavacCompiler.compileInProcess(JavacCompiler.java:596)
[ERROR]         at org.codehaus.plexus.compiler.javac.JavacCompiler.performCompile(JavacCompiler.java:178)
[ERROR]         at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute(AbstractCompilerMojo.java:1134)
[ERROR]         at org.apache.maven.plugin.compiler.CompilerMojo.execute(CompilerMojo.java:187)
[ERROR]         at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:301)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:211)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:165)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:157)
[ERROR]         at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:121)
[ERROR]         at org.mvndaemon.mvnd.builder.SmartBuilderImpl.buildProject(SmartBuilderImpl.java:178)
[ERROR]         at org.mvndaemon.mvnd.builder.SmartBuilderImpl$ProjectBuildTask.run(SmartBuilderImpl.java:198)
[ERROR]         at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:577)
[ERROR]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
[ERROR]         at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
[ERROR]         at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
[ERROR]         at java.base/java.lang.Thread.run(Thread.java:1589)
```

**Comment by elucash** [comment]
this is a bug, we need to "sanitize" the input to `createSourceFile` by unwrapping so called "cached element" to extract original model element (provided by compiler). Should be easy to fix.

**Comment by elucash** [comment]
It appears it might be fixed already, while I'm in the process or releasing 2.9.1, the 2.9.0 release might have the fix for this (albeit it's not recommended due to some memory leak concerns which we address in 2.9.1)

**Comment by dwragge** [comment]
Nice find - this is indeed fixed in 2.9.0. Thanks for the quick follow up! :)

**Comment by elucash** [comment]
2.9.1 was released yesterday, so I definitely recommend the newer version, thanks!
